### ISSUE TITLE
[SDK/Compiler] Sample was broken by new large data passing mechanism

### ISSUE THREAD

**Issue by numerology** [kubeflow/pipelines#2307]
This [Github issue summary sample](https://github.com/kubeflow/examples/blob/master/github_issue_summarization/pipelines/example_pipelines/gh_summ.py) seems to be broken by our new changes.

Reported by Amy 
```
Traceback (most recent call last):
  File "gh_summ1gpu.py", line 114, in <module>
    compiler.Compiler().compile(gh_summ, __file__ + '.tar.gz')
  File "/Users/amyu/miniconda3/envs/mlpipeline2/lib/python3.6/site-packages/kfp/compiler/compiler.py", line 836, in compile
    workflow = self._compile(pipeline_func)
  File "/Users/amyu/miniconda3/envs/mlpipeline2/lib/python3.6/site-packages/kfp/compiler/compiler.py", line 822, in _compile
    return self.create_workflow(pipeline_func=pipeline_func)
  File "/Users/amyu/miniconda3/envs/mlpipeline2/lib/python3.6/site-packages/kfp/compiler/compiler.py", line 813, in create_workflow
    workflow = fix_big_data_passing(workflow)
  File "/Users/amyu/miniconda3/envs/mlpipeline2/lib/python3.6/site-packages/kfp/compiler/_data_passing_rewriter.py", line 146, in fix_big_data_passing
    assert argument_placeholder_parts[2] == 'outputs'
AssertionError
```
@Ark-kun can you take a look when you have time?

**Comment by Ark-kun** [comment]
Duplicate of https://github.com/kubeflow/pipelines/issues/2295
There is a fix PR pending, but there is a license complication.

**Comment by Ark-kun** [comment]
Fixed by #2297
